qa/1589

```diff
@@ -51,21 +53,27 @@ _filter_pcp2json_http()
 
 # real QA test starts here
 port=`_find_free_port`
-nc -l localhost $port >$tmp.nc.out 2>$tmp.nc.err &
-pid1=$!
-sleep 2 # let nc start up
+$PCP_PYTHON_PROG $here/src/pythonserver.py $port &> $tmp.python.out &
+sleep 2
 
-# in case nc(1) does not exit by itself, e.g. on Ubuntu
-( sleep 2; $signal $pid1 ) >>$seq.full 2>&1 &
+pid=$!
+sleep 2 # let server start up
 
 echo "pcp2json invocation" | tee -a $here/$seq.full
-pcp2json -s1 -f%s -ZUTC --url http://localhost:$port/receive hinv.ncpu > $tmp.json.out 2> $tmp.json.err
+pcp2json -s1 -u http://localhost:$port hinv.ncpu >$tmp.json.out 2>$tmp.json.err
+pid32=$!
+sleep 2
+
+kill $pid2
```

Comment:
@natoscott  I wanted to capture the pcp2json pid so I could use kill to act as the "ctrl C" command that pcp2json requires for metrics to output. Is there a different approach that you would take to accomplish that?

Reply:
The -s1 option means "take one sample only" (and then exit) so there should be no need to kill it.

Reply:
@natoscott very odd, running with -s2 in the QA test requires ctrl C but -s1 does not. I guess I will roll with -s1.

Reply:
It will depend on the sample interval I think (-t option) ... -s2 means "take a sample, sleep for sample interval, take a sample, exit" whereas -s1 means "take a sample, exit".